components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -662,23 +676,34 @@ setup_output_file(const std::string& filename,
   set_degrees_of_freedom(filename);
 }
 /* ---------------------------------------------------------- */
+// This routine will evaluate the diagnostics stored in this
+// output instance.
+void AtmosphereOutput::update_field(const std::string& name) const
+{
+  // Check if the field is a diagnostic, if so evaluate it. 
+  if (m_diagnostics.find(name) != m_diagnostics.end()) {
+    const auto& diag = m_diagnostics.at(name);
+    // Check if the diagnostics has any dependencies, if so, evaluate
+    // them as well.  Needed if a diagnostic relies on another
+    // diagnostic.
+    for (const auto& dep : m_diag_depends_on_diags.at(name)) {
+      update_field(dep);
+    }
+    diag->compute_diagnostic();
```

Comment:
I feel like we have to add some book keeping for which diags are computed, in order to avoid computing the same diag twice. This might happen if diag D1 depends on diag D2, and we are asking for both D1 and D2 in our output. In that case, D2 is computed twice: once when processing D2, and once as a dep of D1 when processing D1.

We should keep a simple `std::map<std::string,bool> m_diag_computed` stored in the class. At each run step, we do `m_diag_computed.clear()` right away, so that when we do `m_diag_computed[diag_name]` the first time it returns false. Then, inside this fcn, we update it to true once we're done computing it.

Reply:
why does this need to be a `map`, doesn't `.clear()` just remove everything from the map?  If just being present in the map is enough for us to know it has been computed couldn't be use a `std::vector`?

Reply:
A map is better than a std::vector, since we would access it with a string key. If the key was an int (with small range), then a vector would do.

As for `clear()`, we need to reset all entries to "false" every time we run the output class. We could loop through the map and set all entries to `false`, but clearing the map works just fine (and it's less code), since calling `auto b = map[name];` when `name` is not found would create a new entry, using default ctor (which, for builtin `bool` would init the entry to `false`).

Reply:
okay, thanks for the explanation.  I'll use `map`